### PR TITLE
Adding swagger for endpoint certificates

### PR DESCRIPTION
**Pull request by srmarkov**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ericshape** [comment]
@zhenglaizhang
it seems this is a new Sdk lint error in LintDiff check: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1259820&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=b17d53b8-8799-5fc8-d76a-1d962d18e711

can you help to take a look? thanks!

**Comment by zhenglaizhang** [comment]
It's because of 

> @zhenglaizhang it seems this is a new Sdk lint error in LintDiff check: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1259820&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=b17d53b8-8799-5fc8-d76a-1d962d18e711
> 
> can you help to take a look? thanks!

It's due to at the global parameter space, `BlobAuditingPolicyNameParameter` is duplicated in both `resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json` and `sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/BlobAuditing.json` but with different definition. You can fix it by using the same name or share them if they could be exactly the same.

**Comment by srmarkov** [comment]
> It's because of
> 
> > @zhenglaizhang it seems this is a new Sdk lint error in LintDiff check: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1259820&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=b17d53b8-8799-5fc8-d76a-1d962d18e711
> > can you help to take a look? thanks!
> 
> It's due to at the global parameter space, `BlobAuditingPolicyNameParameter` is duplicated in both `resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json` and `sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/BlobAuditing.json` but with different definition. You can fix it by using the same name or share them if they could be exactly the same.

@zhenglaizhang I addressed the issue you mentioned in this comment, so there are no more LintDiff errors. Still there are 2 SDK errors that seem are not related to my change. Do you have an insight how can I resolve them?

**Review by ericshape — APPROVED** [review]

**Comment by ericshape** [comment]
@zhenglaizhang can you help to take a look about this PR? if there is no other issue, please help to merge it to dev branch? thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+      "description": "The properties of an endpoint certificate.",
+      "type": "object",
+      "properties": {
+        "publicBlob": {
+          "description": "The certificate public blob",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
```
Comment (mentat9) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json:119]: >Resource [](http://example.com/codeflow?start=5&length=8)

Better to use common types definitions for this and ProxyResource definition below. Not blocking from ARM perspective, but definitely a good idea.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EndpointCertificateProperties",
+          "description": "Resource properties.",
```
Comment (mentat9) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json:162]: >Resource properties [](http://example.com/codeflow?start=26&length=19)

More specific description would be good.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
```
Comment (mentat9) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json:187]: >parameters [](http://example.com/codeflow?start=3&length=10)

Most of these are unused and should be removed. Of the ones that are used, SubscriptionIdParameter, ApiVersionParameter, ResourceGroupParameter have counterparts in common types. Recommend using those instead. If you do that and remove the unused ones, the only parameter you need here is ManagedInstanceNameParameter.

**Comment by mentat9** [comment]
Signed off with comments from ARM perspective.

**Review by ruowan — APPROVED** [review]

**Pull request merged**